docs/transpile/custom_passes.mdx

```diff
@@ -0,0 +1,101 @@
+
+```python
+    y_gate_properties = {}
+    for qubit in range(target.num_qubits):        
+        y_gate_properties.update({
+            (qubit,): InstructionProperties(
+                duration = target['x'][(qubit,)].duration,
+                error = target['x'][(qubit,)].error)})
+
+    target.add_instruction(YGate(), y_gate_properties)
+```
+
+Ansatz circuits such as `EfficientSU2` are parameterized, so they much have value bound to them before being sent to the backend. Here we use `.assign_parameters` to attach random values generated by `numpy.random`.
+
+```python
+    qc_t.assign_parameters(np.random.uniform(-np.pi, np.pi, qc_t.num_parameters), inplace=True)
+```
+
+Next, we execute the custom passes. We instantiate the `PassManager` with `ALAPScheduleAnalysis` and `PadDynamicalDecoupling`.  We must run `ALAPScheduleAnalysis` first to add timing information about the quantum circuit before the regularly-spaced dynamical decoupliong sequences can be added. These passes are run on the circuit with `.run()`.
+
+```python
+    pm = PassManager([
+        ALAPScheduleAnalysis(target=target),
+        PadDynamicalDecoupling(target=target, dd_sequence=dd_sequence),
+        ])
+    qc_dd = pm.run(qc_t)
+```
+
+We can use the visualization tool `timeline_drawer` to see the circuit's timing and confirm that a regularly-spaced sequence of `XGate`s and `YGate`s appear in the circuit.
+
+```python
+    from qiskit.visualization import timeline_drawer
+
+    timeline_drawer(qc_dd, show_idle=False)
+```
+![Description](/images/transpile/custom_passes/timeline_dd.png)
+
+Last, because the `YGate` is not an actual basis gate of our backend, we must manually apply the `BasisTranslator` pass (this is a default pass, but it is executed before scheduling, so we need to apply it again). The session equivalence library is a library of circuit equivalences that allows the transpiler to decompose circuits into basis gates, as also specified as an argument.
+
+```python
+    from qiskit.circuit.equivalence_library import SessionEquivalenceLibrary as sel
+
+    qc_dd = BasisTranslator(sel, basis_gates)(qc_dd)
+    qc_dd.draw('mpl', style='iqp', fold=-1, idle_wires=False)
+```
+![Description](/images/transpile/custom_passes/qc_dd_basis.png)
+
+We can see the absence of `YGate`s in our circuit, and we have explicit timing information in the form of `Delay` gates. This transpiled circuit with dynamical decoupling can now be sent to the backend.
```

Comment:
Just to round off this example could you add a code snippet showing how you'd send this circuit on to a backend?

Reply:
Wouldn't that violate the separate nature of the "run" section?

Reply:
We decided to simply link to the run section.